### ISSUE TITLE
zio-macros : generate accessors for "flat" service

### ISSUE THREAD

**Issue by fcabouat** [zio/zio#3217]
Hi,

In https://github.com/zio/zio/issues/3203, there is some discussion to change the target structure of the `@accessible` macro.

Could be a non-goal, but it might be worth considering flexibility in the source structure, allowing accessors generation from "flat" Traits departing from the "module pattern" conventions ?

(Following Discord discussion) 
> @ioleo if I'm not mistaken the issue you reference discusses changing the target structure of the generated accessors ? My question was about the source : my Service isn't named Service and isn't in a wrapping object.
>
> Instead of Clock / Clock.Service, I have a HasClock / Clock naming scheme and I flattened and exploded the "module pattern".
>
> Package object => only contains type aliases, shared models, exceptions and env/zlayer stuff
Package => contains traits of my services + manually written accessors.
Impl subpackage => services implementations.
>
> I've made a small toy example demonstrating this code structuration preferences : https://github.com/fcabouat/zlayer-opinionated

Cheers,

**Comment by DmytroOrlov-flixbus** [comment]
With **DIstage** accessors for "flat" services are much preferable for me (less visual noise/boilerplate):
```
object MyServicePlugin extends PluginDef {
  make[ForwardEndpoints].fromValue(
    new ForwardEndpoints {
      val legacy = IO.succeed {
        base.in("api" / "legacy")
      }
    })
  make[RequestForwarder].fromHas(for {
    backend <- ZIO.service[SttpBackend[Task, ZioStreams with WebSockets]]
    cfg <- ZIO.service[MyServiceConf]
    endpoint <- ForwardEndpoints.legacy
    request = SttpClientInterpreter.toRequest(endpoint, Uri(cfg.uri).some)
  } yield new RequestForwarder {
    def forward(req: MyServiceReq) = backend.send(request(req)).zioBody
  })
}

@accessible
trait ForwardEndpoints {
  def legacy: UIO[Endpoint[MyServiceReq, String, (String, StatusCode), Any]]
}

@accessible
trait RequestForwarder {
  def forward(req: MyServiceReq): ZIO[LogZIO, Capture[TapirErr[String, *]], (String, StatusCode)]
}
```
instead of current:
```
object MyServicePlugin extends PluginDef {
  make[ForwardEndpoints.Service].fromValue(
    new ForwardEndpoints.Service {
      val legacy = IO.succeed {
        base.in("api" / "legacy")
      }
    })
  make[RequestForwarder.Service].fromHas(for {
    backend <- ZIO.service[SttpBackend[Task, ZioStreams with WebSockets]]
    cfg <- ZIO.service[MyServiceConf]
    endpoint <- ForwardEndpoints.legacy
    request = SttpClientInterpreter.toRequest(endpoint, Uri(cfg.uri).some)
  } yield new RequestForwarder.Service {
    def forward(req: MyServiceReq) =
      backend.send(request(req)).zioBody
  })
}

@accessible
object ForwardEndpoints {

  trait Service {
    def legacy: UIO[Endpoint[MyServiceReq, String, (String, StatusCode), Any]]
  }

}

@accessible
object RequestForwarder {

  trait Service {
    def forward(req: MyServiceReq): ZIO[LogZIO, Capture[TapirErr[String, *]], (String, StatusCode)]
  }

}
```
So redundant `make[ForwardEndpoints.Service]...` become just `make[ForwardEndpoints]...`.
And
```
@accessible
object ForwardEndpoints {

  trait Service {
    def legacy: UIO[Endpoint[MyServiceReq, String, (String, StatusCode), Any]]
  }

}
```
become
```
@accessible
trait ForwardEndpoints {
  def legacy: UIO[Endpoint[MyServiceReq, String, (String, StatusCode), Any]]
}
```
https://github.com/zio/zio/pull/5122
https://github.com/zio/zio-intellij/pull/267

**Comment by adamgfraser** [comment]
Closed by #5122.
